Makefile

```diff
@@ -259,19 +259,6 @@ default: all
 WARNING_FLAGS = -W -Wextra -Wall -Wsign-compare -Wshadow \
   -Wno-unused-parameter
 
-CCVERSION = $(shell $(CXX) -dumpversion)
-CCNAME = $(shell $(CXX) --version | awk 'NR==1' | cut -f1 -d " ")
-
-ifeq ($(CCNAME), clang)
-ifeq ($(CCVERSION), 4*)
-	CXXFLAGS += -faligned-new
-endif
-else
-ifeq ($(CCVERSION), 7)
-	CXXFLAGS += -faligned-new
-endif
-endif
```

Comment:
These changes are introduced by #2672 to fix gcc-7 compile error which is caused by mis-aligned of LRUCacheShard. The -faligned-new flag will allow LRUCacheShard to properly aligned with native new[] operator. Since we get a proper fix for the issue, we don't need them anymore.